packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -292,7 +292,16 @@ class OnsetPicker extends React.Component {
 
 
     let granularitySelect = null;
+    const birthMoment = moment(birthdate).startOf('day'); // startOf to clear time from values
+    const currentMoment = moment().startOf('day');
     if (this.state.precision !== PrecisionOptions.UNKNOWN) {
+      const optionAge = (currentMoment.diff(birthMoment, 'weeks') !== 0) ? (
+        <SelectField.Option
+          value={GranularityOptions.AGE}
+          display={intl.formatMessage({ id: 'Terra.onsetPicker.age' })}
+          key={GranularityOptions.AGE}
+        />
+      ) : <React.Fragment />;
```

Comment:
It's a little strange to return a raw fragment here, I think you can return undefined.

Reply:
So, when I put `undefined` instead of a `raw fragment`. At that time the code breaks at this line [SelectUtils]( https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/shared/_SelectUtil.js#L101)

Reply:
The logic may need to change to either conditionally push these into an array or conditionally add it below. I don't think an empty fragment should be added as a workaround, the option should be removed from the list instead of substituted.

Reply:
updated here https://github.com/cerner/terra-clinical/pull/543/commits/0c708f260e59a74b02bcb9a9a6e76e5a2035e4c3.